src/main/java/seedu/realodex/model/person/predicates/TagsMatchPredicate.java

```diff
@@ -0,0 +1,47 @@
+package seedu.realodex.model.person.predicates;
+
+import java.util.Set;
+import java.util.function.Predicate;
+
+import seedu.realodex.commons.util.ToStringBuilder;
+import seedu.realodex.model.person.Person;
+import seedu.realodex.model.tag.Tag;
+
+
+/**
+ * Tests that a {@code Person}'s {@code Name} contains the keyphrase given.
+ */
+public class TagsMatchPredicate implements Predicate<Person> {
+    private final Set<Tag> tagSet;
+
+    public TagsMatchPredicate(Set<Tag> tagSet) {
+        this.tagSet = tagSet;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        Set<Tag> personTags = person.getTags();
+        return personTags.containsAll(tagSet);
+    }
```

Comment:
Good job on this! It works fine for me and covers all corner cases I could think of.